sympy/combinatorics/fp_groups.py

```diff
@@ -503,15 +503,12 @@ def compress(self):
         chi = tuple([i for i in range(len(self.p)) if self.p[i] != i])
         for alpha in self.omega:
             gamma += 1
-            # a non-live coset is found with `gamma ~ alpha`
             if gamma != alpha:
                 # replace α by γ in coset table
                 for x in A:
                     beta = self.table[alpha][A_dict[x]]
-                    if beta == alpha:
```

Comment:
This line introduced the bug, its not required as the below this does the task. I have now verified "Mennickie group" (n=1, n=2), Cox group, "BH Neuman", "Burnside groups" (some more groups) from GAP for exact coset value (not just `index`) and these seem to be working good.